main.py

```diff
@@ -1033,17 +1034,27 @@ def notify_interface_link_status(self, interface, reason):
 
     def notify_link_status_change(self, link, reason='not given'):
         """Send an event to notify about a status change on a link."""
-        name = 'kytos/topology.'
-        if link.status == EntityStatus.UP:
-            status = 'link_up'
+        link_id = link.id
+        if not link.status_reason and link_id in self.link_down:
```

Comment:
We're still implementing `status_reason` gradually, liveness still hasn't implemented it yet. I'll recommend that you rely on `link.status` at this time, if you could refactor here. 

Also, we'll need a thread lock since this is accessed from multiple threads and self.link_down is being read and written to avoid corner cases. 

Another case to simulate would be an interfaces that goes into maintenance, controller restart, and then see if it'll potentially send another link down, if it does this will need to change and self.link_down would need to be initialized with all existing links once topology is loaded or to simplify even more maybe you could only keep track of `link_up` instead? That way if it's not there it's not UP, that seems to facilitate.

Reply:
Looking forward to the local tests and e2e tests outcome later on.